### ISSUE TITLE
Add support for Monte Carlo realizations with precomputed randoms

### ISSUE THREAD

**Issue by aphearin** [ArgonneCPAC/diffstar#123]
DiffstarPop currently accepts a `jax.random.key` in order to generate a SFH population. It would be helpful for the API to include support for using a precomputed set of randoms, e.g., drawn in advance from a multivariate normal with the identity matrix for covariance (for the case of main sequence and quenched sequence) and uniform random (for the case of `mc_is_q`). This API modification would round out support within diffsky for fully pre-computed randoms, which allows for greater control over stochasticity in our gradient descent and HMCs, e.g., when running gradient descent in epochs, or eliminating MC noise in our posteriors.
